### PR TITLE
Prevent modal popup from removing other resize handlers

### PR DESCRIPTION
**Pull request by spadgett**
@rhamilto This should fix the nav problem you were seeing.

### CODE REVIEW

**Comment by spadgett** [comment]
@jeff-phillips-18 If the handler we pass in as the second arg is null, it removes all resize listeners.

**Comment by jeff-phillips-18** [comment]
I see. Probably at init the debounceResize isn't set yet. The $onChanges should probably be checking for first change or that  newval !== oldval before calling onHide.

**Comment by spadgett** [comment]
We also call onHide onDestroy

**Comment by jeff-phillips-18** [comment]
But the shown value should be correct at that point and that would mean the debounceResize is set. I'm not disagreeing with the fix, just thinking of how this went wrong to begin with ;)

**Comment by spadgett** [comment]
> But the shown value should be correct at that point and that would mean the debounceResize is set. 

It looks like that's not resize listener is not added until the dialog becomes visible (which is good). So if the component is destroyed without showing the dialog, it will remove all listeners.

Either way I think it's good to have the guard so we never accidentally remove other listeners.

File: src/components/origin-modal-popup/origin-modal-popup.js [hunk 1]
```diff
@@ -81,7 +81,10 @@ angular.module("openshiftCommonUI").component("originModalPopup", {
 
     function onHide() {
       hideModalBackdrop();
-      angular.element($window).off('resize', debounceResize);
+      if (debounceResize) {
+        angular.element($window).off('resize', debounceResize);
+        debounceResize = null;
```
Comment (spadgett) [inline src/components/origin-modal-popup/origin-modal-popup.js]: @jeff-phillips-18 I updated the PR to also null out `debounceResize`

**Pull request merged**